packages/components/tests/e2e/specs/vsCodeExtension.spec.js

```diff
@@ -747,6 +747,35 @@ context('VS Code Extension', () => {
       cy.get('.trace-node[data-event-id="15"]')
         .should('be.visible')
         .should('have.class', 'highlight');
+
+      cy.get('.trace-node[data-event-id="13"]')
+        .click()
+        .should('have.class', 'highlight');
+      cy.get('.trace-node[data-event-id="1"]').should(
+        'not.have.class',
+        'filtered'
+      );
+      cy.get('.trace-node[data-event-id="3"]').should(
+        'not.have.class',
+        'filtered'
+      );
+      cy.get('.trace-node[data-event-id="15"]').should(
+        'not.have.class',
+        'highlight'
+      );
+
+      cy.get('.details-panel__buttons').contains('Clear selection').click();
+
+      cy.get('.trace-node[data-event-id="13"]').should(
+        'not.have.class',
+        'highlight'
+      );
+      cy.get('.trace-node[data-event-id="1"]').should('have.class', 'filtered');
+      cy.get('.trace-node[data-event-id="3"]').should('have.class', 'filtered');
+      cy.get('.trace-node[data-event-id="15"]').should(
+        'have.class',
+        'highlight'
+      );
```

Comment:
There should also be a test where the next (and previous) button is clicked, and the active selection changes to the _nearest_ filtered event in that direction.

In other words, clicking the left search arrow moves to the nearest previous filtered event. The right search arrow moves to the nearest next filtered event.

![image](https://user-images.githubusercontent.com/8737782/149215164-2778b431-ed31-4eca-a376-8cbcc433d019.png)

Reply:
Done https://github.com/applandinc/appmap-js/pull/519/commits/bdd8356bfa37b513a18ea9862d9deb28a2720586

Now all filtered events are ordered by ID, and I changed test input filter to verify that behavior:
https://github.com/applandinc/appmap-js/pull/519/commits/bdd8356bfa37b513a18ea9862d9deb28a2720586#diff-f5e13ae703840c989eb60ca988f5d4e016a49c59a6e4366c0cdeead459043f88R710

Reply:
This is working much nicer than before. I think an ideal UX would be to select the nearest filtered events to the current selection, regardless of what the last selected filtered event index was.

As an example:
![image](https://user-images.githubusercontent.com/8737782/150396052-3b6b246c-b828-4c08-97d8-e23adf731108.png)

`ActiveSupport::SecurityUtils.secure_compare` is not part of the filtered set. If I click the previous or next filtered event button, I'd expect to always select one of the two `JSON::Ext::Parser#parse` events shown here as appropriate. This would be regardless of any previous filtered event index I had selected.

Let's save this for a future story, because the current behavior is still a much needed improvement over what we currently have.

Reply:
@dustinbyrne would you mind to create new issue for the future?